app/services/rdf_importer.rb

```diff
@@ -0,0 +1,52 @@
+class RdfImporter
+  attr_reader :url, :errors, :term_list
+  delegate :url_to_graph, :graph_to_termlist, :validators, :to => :injector
+
+  def initialize(errors)
+    @errors = errors
+  end
+
+  def call(url)
+    reset(url)
+    validators.each {|v| v.new.validate(self)}
+    build_graph
+    build_term_list
+  end
+
+  private
+
+  def reset(url)
+    errors.clear
```

Comment:
The validation/error situation gets really ugly if you need to just change a value (such as URL) and re-check validation status.  In a Rails object, valid? resets errors for you, and I was concerned that `call` with a different URL shouldn't hold onto old errors.

I actually wanted RdfImporter to have its own errors and make ImportForm responsible for propagating them up, but `ActiveModel::Error.new` can't be called without an ActiveModel-like object (it does stuff when you add errors, such as calling model_name).

That said, I agree that this needs to go.  I'm just not yet sure how.